CommunityFridgeMapApi/dependencies/python/db.py

```diff
@@ -11,4 +11,158 @@ def get_ddb_connection(env:str=os.getenv('Environment', '')) -> 'botocore.client
+        self.food_restrictions:list = fridge.get('food_restrictions', None)
+        self.lat:str = fridge.get('lat', None)
+        self.long:str = fridge.get('long', None)
+        self.profile_image:str = fridge.get('profile_image', None)
+        self.last_editted:int = fridge.get('last_editted', None)
+        self.check_in_time:int = fridge.get('check_in_time', None)
+        self.check_in_notes:str = fridge.get('check_in_notes', None)
+        self.check_in_status:str = fridge.get('check_in_status', None)
+        self.check_in_image:str = fridge.get('check_in_image', None)
+        self.db_client = db_client
+        if self.fridge_state is not None:
+            self.fridge_state = self.fridge_state.upper()
+
+    def has_required_fields(self) -> bool:
+        for field in self.REQUIRED_FIELDS:
+            if getattr(self, field) is None:
+                raise RequiredFieldMissingException(object_name='fridge', field=field)
+        return True
+
+    def is_valid_fridge_state(self) -> bool:
+        if self.fridge_state not in self.STATES:
+            raise InvalidStateException(fridge_state=self.fridge_state)
+        return True
+
+    def get_fridge(db_client):
+        pass
+
+    def get_fridges(db_client):
+        pass
+
+    def add_fridges(db_client):
+        pass
+    
+    def set_username(self):
+        #Fridge username is the display_name with no spaces and all lower cased
+        username = self.display_name.lower().replace(" ", "")
+        if not username.isalnum():
+            raise InvalidDisplayNameException(username)
+        self.username = username
+
+
+    def build_fridge_item(self):
+        #generates a dictionary in the format dynamodb expects
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.put_item
+        fridge_item = {}
+        for key in self.FRIDGE_TYPES:
+            val = getattr(self, key)
+            if val is not None:
+                fridge_item[key] = {self.FRIDGE_TYPES[key]: val}
+        return fridge_item
+
+
+    def add_fridge(self):
+        self.has_required_fields()
+        self.is_valid_fridge_state()
+        self.set_username()
+        item = self.build_fridge_item()
+        conditional_expression = 'attribute_not_exists(fridge_state) AND attribute_not_exists(username)'
+        response  = self.db_client.put_item(
+            TableName=self.TABLE_NAME,
+            Item=item,
+            ConditionExpression=conditional_expression
+        )
+        return response
+
+    def get_fridge_locations(db_client):
+        pass
+
+    def get_all_fridges(db_client):
+        return db_client.scan(TableName=Fridge.TABLE_NAME)
+
+
+class FrigeCheckIn:
+
+    def __init__(self):
+        pass
+
+class FridgeHistory:
+    
+    def __init__(self):
+        pass
+
+class Tag:
+
+    def __init__(self):
+        pass
+
+class RequiredFieldMissingException(Exception):
```

Comment:
Not 100% sure about this, but since we're inheriting a class here, should we also call `super.__init__()`?

Reply:
Yes! If we wanted to use the methods that Exception has. I ended up deleting the custom exception though, so this no longer applies